perceval/backends/core/rocketchat.py

```diff
@@ -0,0 +1,338 @@
+            yield message
+            nmsgs += 1
+
+        logger.info("Fetch process completed: %s message fetched", nmsgs)
+
+    @staticmethod
+    def parse_room_info(raw_room_info):
+        """Parse a room info JSON stream.
+        This method parses a JSON stream, containing the information
+        from a room, and returns a dict with the parsed data.
+        :param raw_room_info
+        :returns: a dict with the parsed information about a room
+        """
+        result = json.loads(raw_room_info)
+        return result['channel']
+
+    @staticmethod
+    def parse_history(raw_history):
+        """Parse a room history JSON stream.
+        This method parses a JSON stream, containing the history of
+        a room, and returns a list with the parsed data.
+        :param raw_history: JSON string to parse
+        :returns: a tuple with a list of dicts with the parsed messages"""
+        result = json.loads(raw_history)
+        return result['messages']
+
+    @staticmethod
+    def parse_user(raw_user):
+        """Parse a user's info JSON stream.
+        This method parses a JSON stream, containing the information
+        from a user, and returns a dict with the parsed data.
+        :param raw_user: JSON string to parse
+        :returns: a dict with the parsed user's information
+        """
+        result = json.loads(raw_user)
+        return result['u']
+
+    @staticmethod
+    def metadata_category(item):
+        """Extracts the category from a Rocket chat item.
+        This backend only generates one type of item which is
+        'message'.
+        """
+        return CATEGORY_MESSAGE
+
+    @staticmethod
+    def metadata_id(item):
+        """Extracts the identifier from a RocketChat item."""
+        return item["_id"]
+
+    @staticmethod
+    def metadata_updated_on(item):
+        """Extracts the update time from a RocketChat item.
+        The timestamp is extracted from 'ts' field,
+        and sliced to get the year, month and date.
+        It is then converted into a UNIX timestamp.
+        :param item: item generated by the backend
+        :returns: extracted timestamp
+        """
+        ts = item['ts'][:10]
+        ts = time.mktime(datetime.datetime.strptime(ts, "%Y-%m-%d").timetuple())
+
+        return ts
+
+    def _init_client(self, from_archive=False):
+        """Init client"""
+
+        return RocketChatClient(self.url, self.user_id, self.api_token, self.max_items, self.archive,
+                                from_archive, self.ssl_verify, roomId=self.roomId)
+
+    def __get_or_fetch_user(self, user_id, username, name):
+        if user_id in self._users:
+            return self._users[user_id]
+
+        logger.debug("User %s not found on client cache; fetching it", user_id)
+
+        user = {
+            "username": username,
+            "name": name,
+        }
+
+        self._users[user_id] = user
+        return user
+
+
+class RocketChatClient(HttpClient):
```

Comment:
This class should also use the `RateLimitHandler` (example at https://github.com/chaoss/grimoirelab-perceval/blob/master/perceval/backends/core/github.py#L571). This class is used to handle rate limit information, which are returned in the headers of the response (see figure below):

![captura_269](https://user-images.githubusercontent.com/6515067/77090423-6a5cbf80-6a07-11ea-8d49-d58765d85afa.png)

Reply:
What exactly do I need to add in it? 
GitHub supports multiple tokens but I have not yet added the support for multiple API tokens.